### PR TITLE
DOC-4185 Document manual bootstrapping

### PR DESCRIPTION
**Pull request by davidcurrie**

### CODE REVIEW

**Review by yarivus — DISMISSED** [review]
LGTM

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 1]
```diff
@@ -1,13 +1,215 @@
...
+bootstrap:
+  db:
+    enabled: false
+```
+
+The following steps can then be performed to manually set up the database prior to Corda installation.
+
+### Create the Database Schemas
+
+The following Corda CLI command generates DML files in the directory `/tmp/db` for creating the database schema:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:98]:
```suggestion
2. Use the Corda CLI to generate DML files for creating the database schema. For example, the following command generates the files in the directory `/tmp/db`:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 2]
```diff
@@ -1,13 +1,215 @@
...
+
+```shell
+corda-cli.sh database spec -c -l /tmp/db
+```
+
+The DML generated should be reviewed and then executed against the database.
+
+### Create RBAC Database Connection Configuration
+
+The following Corda CLI command generates DDL for populating the RBAC database connection configuration:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:108]:
```suggestion
4. Execute the following Corda CLI command to generate DDL for populating the RBAC database connection configuration:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 3]
```diff
@@ -1,13 +1,215 @@
...
+
+An example usage might look as follows:
+
+```shell
+corda-cli.sh initial-config create-db-config -u rbac-user -p rc9VLHU3 \
+  --name corda-rbac --jdbcURL jdbc:postgresql://postgres.example.com:5432/cordacluster \
+  --jdbcPoolMaxSize 5 --salt X3UaCpUH --passphrase UUWLhD8S -l /tmp/db
+```
+
+The DDL generated should be reviewed and then executed against the database.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:134]:
```suggestion
5. Review the DDL files generated and then execute against the database.
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 4]
```diff
@@ -1,13 +1,215 @@
...
+These must match the values specified in the Corda deployment configuration for the DB worker:
+
+```yaml
+workers:
+  db:
+    salt: <SALT>
+    passphrase: <PASSPHRASE>
+```
+
+An example usage might look as follows:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:126]:
```suggestion
For example:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 5]
```diff
@@ -1,13 +1,215 @@
...
+
+```shell
+corda-cli.sh initial-config create-db-config -u rbac-user -p rc9VLHU3 \
+  --name corda-rbac --jdbcURL jdbc:postgresql://postgres.example.com:5432/cordacluster \
+  --jdbcPoolMaxSize 5 --salt X3UaCpUH --passphrase UUWLhD8S -l /tmp/db
+```
+
+The DDL generated should be reviewed and then executed against the database.
+
+### Create Crypto Database Connection Configuration
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:136]:
```suggestion

```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 6]
```diff
@@ -1,13 +1,215 @@
...
+
+The following Corda CLI command generates DDL for populating the Crypto database connection configuration:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:138]:
```suggestion
6. Execute the following Corda CLI command to generate DDL for populating the Crypto database connection configuration:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 7]
```diff
@@ -1,13 +1,215 @@
...
+
+```shell
+corda-cli.sh initial-config create-db-config -u <CRYPTO-USERNAME> -p <CRYPTO-PASSWORD> \
+  --name corda-crypto --jdbcURL jdbc:postgresql://<DB-HOST>:<DB-PORT>/<DB=NAME>?currentSchema=CRYPTO \
+  --jdbcPoolMaxSize <POOL-SIZE> --salt <SALT> --passphrase <PASSPHRASE> -l /tmp/db
+```
+
+The `<SALT>` and `<PASSPHRASE>` must match those used above and specified in the Corda deployment configuration.
+
+An example usage might look as follows:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:148]:
```suggestion
For example:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 8]
```diff
@@ -1,13 +1,215 @@
...
+
+```shell
+corda-cli.sh initial-config create-db-config -u crypto-user -p TqoCp4v2 \
+  --name corda-crypto --jdbcURL jdbc:postgresql://postgres.example.com:5432/cordacluster?currentSchema=CRYPTO \
+  --jdbcPoolMaxSize 5 --salt X3UaCpUH --passphrase UUWLhD8S -l /tmp/db
+```
+
+The DDL generated should be reviewed and then executed against the database.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:156]:
```suggestion
7. Review the DDL files generated and then execute against the database.
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 9]
```diff
@@ -1,13 +1,215 @@
...
+
+### Create the Initial Admin User
+
+The following Corda CLI command generates DDL for populating the initial admin user for Corda:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:160]:
```suggestion
8.Execute the following Corda CLI command to generate DDL for populating the initial admin user for Corda:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 10]
```diff
@@ -1,13 +1,215 @@
...
+
+```shell
+corda-cli.sh initial-config create-db-config -u crypto-user -p TqoCp4v2 \
+  --name corda-crypto --jdbcURL jdbc:postgresql://postgres.example.com:5432/cordacluster?currentSchema=CRYPTO \
+  --jdbcPoolMaxSize 5 --salt X3UaCpUH --passphrase UUWLhD8S -l /tmp/db
+```
+
+The DDL generated should be reviewed and then executed against the database.
+
+### Create the Initial Admin User
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:158]:
```suggestion

```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 11]
```diff
@@ -1,13 +1,215 @@
...
+
+The following Corda CLI command generates DDL for populating the initial admin user for Corda:
+
+```shell
+corda-cli.sh initial-config create-user-config -u <INITIAL-ADMIN-USERNAME> -p <INITIAL-ADMIN-PASSWORD> -l /tmp/db
+```
+
+The DDL generated should be reviewed and then executed against the database.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:166]:
```suggestion
9. Review the DDL files generated and then execute against the database.
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 12]
```diff
@@ -1,13 +1,215 @@
...
+
+### Create the Database Users and Grant Access
+
+Create the RBAC and Crypto users and grant access as follows:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:170]:
```suggestion
10. Create the RBAC and Crypto users and grant access as follows:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 13]
```diff
@@ -1,13 +1,215 @@
...
+
+The following Corda CLI command generates DDL for populating the initial admin user for Corda:
+
+```shell
+corda-cli.sh initial-config create-user-config -u <INITIAL-ADMIN-USERNAME> -p <INITIAL-ADMIN-PASSWORD> -l /tmp/db
+```
+
+The DDL generated should be reviewed and then executed against the database.
+
+### Create the Database Users and Grant Access
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:168]:
```suggestion

```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 14]
```diff
@@ -1,13 +1,215 @@
...
+```sql
+CREATE USER <RBAC-USERNAME> WITH ENCRYPTED PASSWORD '<RBAC-PASSWORD>';
+GRANT USAGE ON SCHEMA RPC_RBAC to <RBAC-USERNAME>;
+GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA RPC_RBAC to <RBAC-USERNAME>;
+CREATE USER <CRYPTO-USERNAME> WITH ENCRYPTED PASSWORD '<CRYPTO-PASSWORD>';
+GRANT USAGE ON SCHEMA CRYPTO to <CRYPTO-USERNAME>;
+GRANT SELECT, INSERT, UPDATE, DELETE ON ALL TABLES IN SCHEMA CRYPTO to <CRYPTO-USERNAME>;
+```
+
+### Create the Initial Crypto Configuration
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:181]:
```suggestion

```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 15]
```diff
@@ -1,13 +1,215 @@
...
+
+The following Corda CLI command generates DDL for populating the initial crypto configuration:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:183]:
```suggestion
11. Execute the following Corda CLI command to generate DDL for populating the initial crypto configuration:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 16]
```diff
@@ -1,13 +1,215 @@
...
+
+```shell
+corda-cli.sh initial-config create-crypto-config --salt <SALT> --passphrase <PASSPHRASE> -l /tmp/db
+```
+
+The `<SALT>` and `<PASSPHRASE>` must match those used above and specified in the Corda deployment configuration.
+
+The DDL generated should be reviewed and then executed against the database.
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:191]:
```suggestion
12. Review the DDL files generated and then execute against the database.
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 17]
```diff
@@ -1,13 +1,215 @@
...
+
+## RBAC Roles
+
+A post-install job normally creates three default RBAC roles for the Corda API.
+Automatic creation of these RBAC roles can be disabled as follows:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:196]:
```suggestion
1. Set the following override in the deployment configuration to disable the automatic creation:
```

File: content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md [hunk 18]
```diff
@@ -1,13 +1,215 @@
...
+
+```yaml
+bootstrap:
+  rbac:
+    enabled: false
+```
+
+To manually create the three RBAC roles, execute the following three commands:
```
Comment (nadinequinr3) [inline content/en/platform/corda/5.0-alpha/deploying/deployment-tutorials/manual.md:204]:
```suggestion
2. Execute the following three commands:
```

**Review by nadinequinr3 — CHANGES_REQUESTED** [review]

**Review by nadinequinr3 — DISMISSED** [review]

**Review by nadinequinr3 — APPROVED** [review]

**Review by yarivus — COMMENTED** [review]
Thanks @nadinequinr3

**Pull request merged**